### ISSUE TITLE
Integrate CDK to Takomo

### ISSUE THREAD

**Issue by hmeltaus** [takomo-io/takomo#128]
It would be cool to integrate CDK somehow to Takomo.

**Comment by mfonsen** [comment]
I noticed that the Sceptre project has now CDK support: https://docs.sceptre-project.org/dev/docs/templates.html#aws-cdk (which points to https://github.com/sceptre/sceptre-cdk-handler). 

The following quote from the repository readme pretty much makes sense also for Takomo:

> 90% of everything CDK does is at "compile-time", when the templates are being rendered. It has a very limited ability to execute custom code at deployment time, and never really between the deployments of custom stacks and resources. Furthermore, its way of wiring together stacks uses CloudFormation exported outputs and imports, which are rather rigid and can have a host of unintended consequences, especially when values might change.

>In contrast, Sceptre is a deployment orchestration tool. It excels in deploying entire environments of stacks, "wiring them together"

The key benefit from using Takomo could be native support for Typescript whereas Sceptre supports tight integration with Python defined CDK classes.

**Comment by hmeltaus** [comment]
Yeah, I've been working with CDK a lot lately and can agree with the Sceptre team's findings. I guess I should revisit the idea of integrating CDK now that I have gained more understanding of its workings.

**Comment by mfonsen** [comment]
The AWS CDK Toolkit Library is now in GA: https://community.aws/content/2wm6TNpUlPMVgcvXVySywxWaO7T/build-custom-cli-s-deployment-automation-and-more-with-the-aws-cdk-toolkit-library.

Based on my early experience use of CDK deployments from code is now easier and a faster. The Toolkit Library has been designed to support deployment of multiple applications to multiple environments in parallel.

**Comment by hmeltaus** [comment]
Hi, the AWS CDK Toolkit Library looks great! I’ll need to take a closer look at it.

As for adding CDK integration into Takomo, I’ll skip that. It would take quite a bit of effort, and since the AWS CDK Toolkit Library already provides the needed functionality, it makes more sense to rely on that. I’d also like to keep Takomo lightweight as a straightforward CloudFormation wrapper.

**Comment by hmeltaus** [comment]
We won’t proceed with implementing this, because it would require a lot of work and the same can already be achieved with existing tools.
